libcaf_net/test/datagram_socket.cpp

```diff
@@ -0,0 +1,35 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright 2011-2019 Dominik Charousset                                     *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#define CAF_SUITE datagram_socket
+
+#include "caf/net/datagram_socket.hpp"
+
+#include "caf/test/dsl.hpp"
+
+using namespace caf;
+using namespace caf::net;
+
+CAF_TEST(invalid_socket) {
+  datagram_socket x;
+#ifdef CAF_WINDOWS
+  CAF_CHECK_EQUAL(allow_connreset(x, true), sec::network_syscall_failed);
+#else  // CAF_WINDOWS
+  CAF_CHECK_EQUAL(allow_connreset(x, true), none);
```

Comment:
I think this should fail on non-Windows systems as well. Simply because passing `invalid_socket` should get checked (e.g. see `child_process_inherit` on Windows). Then we can check for `CAF_CHECK_NOT_EQUAL(..., none);`.